### ISSUE TITLE
Todo-method

### ISSUE THREAD

**Issue by hcatlin** [sass/sass-spec#177]
I know we keep discussing various meta-comment-style-shit things to help track issues. But, one of the most annoying things right now is that the "todo" flag only applies to libsass and isn't setup for multiple projects.

How about we drop a file in todo-tests called ```libsass.todo```. So, your test folder ```/extends/basic``` might include a new file called ```rubysass.todo``` and that would indicate to the ruby people that it's still a  known breaking test for them.

**Comment by chriseppstein** [comment]
There is other metadata that needs to be tracked as well:

1. `since_version` - The first language version for which the test applies.
2. `until_version` - The last language version for which the test applies.
3. `features` - Some tests are about how features interact and don't apply to a single feature. We might want to track the "other features" a test applies to.

The file approach is pretty good for simple flags but not as well suited for key/value pairs. Perhaps a yaml or json file like `details.json` where we can track other information about each test in an easily parsable format? This would also make it easier to add human explanations without resorting to comments for everything.

**Comment by xzyfer** [comment]
Yaml front matter was once suggested as a spec format. It was interesting
because it meant the spec and it's config where in the one file and it was
easily parsable.

**Comment by mgreter** [comment]
I currently use a comma separated list which supports all options found in the C-API. But I did not add anything in regard to `since_version`. I just added support for some marker files to skip certain tests for libsass. But I agree that this would be a very nice options to have (this would pratically mean to have a "version" specific to the spec suite and not the engine itself). One idea I had was to add multiple output files to the spec, like so:

```
input.scss
output.css
output.3.2.css
output.3.1.css
````

or

````
input.scss
output.css
output.libsass.css
output.3.2.css
output.libsass.3.1.css
output.libsass.3.0.skip
````

Also check out https://github.com/sass/sass-spec/issues/131, where I "added" support to test all output-styles. This could of cource be combined with this approach. One can see that this gets complex pretty easy :wink: We could also add an option/flag wether to test for white-space or not?

This is really just an idea and I didn't think this through! IMO the use of all those specific output files should not be needed too often, but there are some edge cases which I'm not sure if we can get correct in libsass (somethimes it even feels like a ruby sass bug). Take this example in compressed output style:

```
color: #00FF00;
ruby -> color: #0f0
libsass -> color: lime
```

The actual edge case is a bit more complex (AFAIK this is the case if the color comes from a variable). But IMHO libsass is correct here, since `lime` and `#0f0` have the same length. Sometimes ruby sass also outputs `lime`, but I wasn't able to figure out the logic behind it. This is all work in progress so I didn't yet have fully verify this. Otherwise I already would have opened a issue for that.
